### PR TITLE
Added rowStyle prop to SimpleList

### LINKED ISSUE #5253: Make SimpleList compatible with rowStyle property.
**Is your feature request related to a problem? Please describe.**
Using the `rowStyle `property on the `Datagrid `component is great. But when you switch to the responsive `SimpleList `component you suddenly loose the ability to style each row based on the `record `values. 

The way SimpleList is written right now allows absolutely no way to plug in record-based conditional styling.

**Describe the solution you'd like**
Adding the prop `rowStyle `to the `SimpleList `component which could be applied to the iterating `ListItem` to allow full styling through nested selectors.

**Describe alternatives you've considered**

Here is a PR #5252

### PR DESCRIPTION
**Pull request by ValentinnDimitroff**
Closes #5253

### CODE REVIEW

File: packages/ra-ui-materialui/src/list/SimpleList.tsx [hunk 1]
```diff
@@ -95,15 +96,18 @@ const SimpleList: FC<SimpleListProps> = props => {
...
                         linkType={linkType}
                         basePath={basePath}
                         id={id}
                         key={id}
                         record={data[id]}
                     >
-                        <ListItem button={!!linkType as any}>
+                        <ListItem
+                            button={!!linkType as any}
+                            style={rowStyle ? rowStyle(data[id], rowIndex) : null}
```
Comment (jdemangeon) [inline packages/ra-ui-materialui/src/list/SimpleList.tsx:109]: I'd prefer using `undefined` instead of `null` in this case.

**Review by jdemangeon — CHANGES_REQUESTED** [review]
Thanks for your PR! Can you rebase on `next` and make test pass please?

**Comment by djhi** [comment]
Hey, thanks for the pull request. Would you mind to also run the prettier command (`yarn prettier`) to fix the CI build?

File: packages/ra-ui-materialui/src/list/SimpleList.tsx (hunk 1 above)
Comment (ValentinnDimitroff) [reply packages/ra-ui-materialui/src/list/SimpleList.tsx:109]: Sure, will change that!

**Comment by ValentinnDimitroff** [comment]
I have ran the `yarn prettier` and made a new PR to the `next `branch.

**Review by djhi — CHANGES_REQUESTED** [review]
Can you please revert the two unrelated changes (cypress.json and package.json)?

**Comment by ValentinnDimitroff** [comment]
Ok, sorry guys for the other PR, I got little bit confused with you asked me to do but I think now it should be ok?

I rebased to the `next `branch. Still got CI issues, I don't get it why.

@djhi Yes, I will revert those changes. I think they came after I ran the `yarn prettier`.

**Comment by djhi** [comment]
> Yes, I will revert those changes. I think they came after I ran the yarn prettier.

Ah sorry about that, I'll take a loot at its configuration. Thanks for reporting it!

**Comment by ValentinnDimitroff** [comment]
It's fine, I am a newly contributor so still getting to grips. If some other changes are needed just ping me and I will update the branch! :)

**Comment by djhi** [comment]
Ignore my comments on the unrelated files. It is indeed prettier related and it's actually right.

File: packages/ra-ui-materialui/src/list/SimpleList.tsx [hunk 2]
```diff
@@ -95,15 +96,20 @@ const SimpleList: FC<SimpleListProps> = props => {
...
                         basePath={basePath}
                         id={id}
                         key={id}
                         record={data[id]}
                     >
-                        <ListItem button={!!linkType as any}>
+                        <ListItem
+                            button={!!linkType as any}
+                            style={
+                                rowStyle ? rowStyle(data[id], rowIndex) : null
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/SimpleList.tsx:110]: It seems you didn't replace `null` with `undefined` as requested by @jdemangeon. Can you please fix that?

**Review by djhi — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/list/SimpleList.tsx (hunk 2 above)
Comment (ValentinnDimitroff) [reply packages/ra-ui-materialui/src/list/SimpleList.tsx]: I have but accidentially reverted this one also. Brought it back again, sorry.

**Review by djhi — APPROVED** [review]
Thanks!

**Comment by djhi** [comment]
Sorry again, would you mind documenting the feature as well ? The file to update is https://github.com/marmelab/react-admin/blob/master/docs/List.md#the-simplelist-component

**Comment by ValentinnDimitroff** [comment]
> 
> 
> Sorry again, would you mind documenting the feature as well ? The file to update is https://github.com/marmelab/react-admin/blob/master/docs/List.md#the-simplelist-component

I was actually already doing this :) No worries - will update in a while.

**Comment by fzaninotto** [comment]
You said you rebased your PR on `next`, yet you opened it against `master`. Can you change the PR destination?

**Comment by ValentinnDimitroff** [comment]
> 
> 
> You said you rebased your PR on `next`, yet you opened it against `master`. Can you change the PR destination?

Thanks for the patience - finally did it the right way I hope!

**Comment by fzaninotto** [comment]
Yep, the PR is on the right branch now. We're just waiting for your documentation update.

**Comment by ValentinnDimitroff** [comment]
I added just a new row in the table with props as it is well described in the Datagrid section. Is it needed to provide again a usage example?

**Comment by djhi** [comment]
> Is it needed to provide again a usage example?

Iw ould be way better :+1: Thanks for taking the time to complete this!

**Comment by ValentinnDimitroff** [comment]
Sorry for the late reply - I have added the usage of the prop with editing the existing example in order to keep the page tight and clean. Also tried to stick to the Datagrid example reference. If anything is missing just comment and I'll polish it till the end :) (responding quite faster from now on)

**Review by djhi — APPROVED** [review]
:+1: Thanks a lot!

**Pull request merged**